modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/CheckboxTag.java

```diff
@@ -0,0 +1,170 @@
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.clay.servlet.taglib;
+
+import com.liferay.frontend.taglib.clay.internal.servlet.taglib.BaseContainerTag;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.util.Set;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Kresimir Coko
+ */
+public class CheckboxTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		return super.doStartTag();
+	}
+
+	public boolean getChecked() {
+		return _checked;
+	}
+
+	public boolean getDisabled() {
+		return _disabled;
+	}
+
+	public boolean getIndeterminate() {
+		return _indeterminate;
+	}
+
+	public String getLabel() {
+		return _label;
+	}
+
+	public String getName() {
+		return _name;
+	}
+
+	public boolean getShowLabel() {
+		return _showLabel;
+	}
+
+	public String getValue() {
+		return _value;
+	}
+
+	public void setChecked(boolean checked) {
+		_checked = checked;
+	}
+
+	public void setDisabled(boolean disabled) {
+		_disabled = disabled;
+	}
+
+	public void setIndeterminate(boolean indeterminate) {
+		_indeterminate = indeterminate;
+	}
+
+	public void setLabel(String label) {
+		_label = label;
+	}
+
+	public void setName(String name) {
+		_name = name;
+	}
+
+	public void setShowLabel(boolean showLabel) {
+		_showLabel = showLabel;
+	}
+
+	public void setShowLabel(String value) {
+		_value = value;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_checked = false;
+		_disabled = false;
+		_indeterminate = false;
+		_label = null;
+		_name = null;
+		_showLabel = true;
+		_value = null;
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
+		cssClasses.add("custom-control");
+		cssClasses.add("custom-checkbox");
+
+		return super.processCssClasses(cssClasses);
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		jspWriter.write("<label><input");
+
+		if (_checked) {
+			jspWriter.write(" checked");
+		}
+
+		jspWriter.write(" class=\"custom-control-input\"");
+
+		if (_disabled) {
+			jspWriter.write(" disabled");
+		}
+
+		if (Validator.isNotNull(_name)) {
+			jspWriter.write(" name=\"");
+			jspWriter.write(_name);
+			jspWriter.write("\"");
+		}
+
+		if (Validator.isNotNull(_value)) {
+			jspWriter.write(" value=\"");
+			jspWriter.write(_value);
+			jspWriter.write("\"");
+		}
+
+		jspWriter.write(" type=\"checkbox\" />");
+
+		jspWriter.write("<span class=\"custom-control-label\"><span");
+		jspWriter.write(" class=\"custom-control-label-text");
+
+		if (!_showLabel) {
+			jspWriter.write(" sr-only");
+		}
+
+		jspWriter.write("\">");
+		jspWriter.write(_label);
+		jspWriter.write("</label></label>");
```

Comment:
`</label></label>` doesn't feel right 🤔

Reply:
It's because I used `label` instead of `span`, here's how it's in RadioTag: `jspWriter.write("</span></span></label>");`

Reply:
I read up on it just a little bit, and found out that wrapping it like this means that we don't need an `id` on `input` and `for` on `label`, it doesn't mean that we shouldn't use a semantic `label` instead of generic `span` for the actual label. Please correct me if I'm wrong.
```
<label>
    <input />
    <label />
</label>
```

Reply:
I'm not sure nesting `label` tags is semantic. Based on the spec, `<label>` accepts:

> Phrasing content, but with no descendant labelable elements unless it is the element's labeled control, and **no descendant label elements**.

Reply:
In case it's not clear, I think nesting `<label>` elements is not valid HTML.

Reply:
Yea it is clear, I'll give it a quick look-over just so it sticks to my brain 🧠